src/main/kotlin/org/neo4j/graphql/Predicates.kt

```diff
@@ -86,18 +97,34 @@ data class ExpressionPredicate(val name: String, val op: Operators, val value: A
 }
 
 
-data class RelationPredicate(val fieldName: String, val op: Operators, val value: Map<*, *>, val type: GraphQLFieldsContainer) : Predicate {
-    val not = if (op.not) "NOT" else ""
+data class RelationPredicate(val fieldName: String, val op: RelationOperator, val value: Map<*, *>, val type: GraphQLFieldsContainer) : Predicate {
+    companion object {
+        private val LOGGER = LoggerFactory.getLogger(RelationPredicate::class.java)
+    }
+
+    val not = if (op == RelationOperator.NOT) "NOT" else ""
     // (type)-[:TYPE]->(related) | pred] = 0/1/ > 0 | =
     // ALL/ANY/NONE/SINGLE(p in (type)-[:TYPE]->() WHERE pred(last(nodes(p)))
     // ALL/ANY/NONE/SINGLE(x IN [(type)-[:TYPE]->(o) | pred(o)] WHERE x)
 
     override fun toExpression(variable: String): Cypher {
         val prefix = when (op) {
-            Operators.EQ -> "ALL"
-            Operators.NEQ -> "ALL" // bc of not
+            RelationOperator.EQ_OR_NOT_EXISTS -> "ALL"
+            RelationOperator.NOT -> "ALL" // bc of not
             else -> op.op
         }
+        if (type.getFieldDefinition(fieldName).isList()) {
+            if (op == RelationOperator.EQ_OR_NOT_EXISTS) {
+                LOGGER.info("$fieldName on type ${type.name} was used for filtering, consider using ${fieldName}${RelationOperator.EVERY.suffix} instead")
+            }
+        } else {
+            @Suppress("NON_EXHAUSTIVE_WHEN")
+            when (op) {
+                RelationOperator.SINGLE -> LOGGER.warn("Using $fieldName${RelationOperator.SINGLE.suffix} on type ${type.name} is deprecated, use ${fieldName} directly")
+                RelationOperator.SOME -> LOGGER.warn("Using $fieldName${RelationOperator.SOME.suffix} on type ${type.name} is deprecated, use ${fieldName} directly")
+                RelationOperator.NONE -> LOGGER.warn("Using $fieldName${RelationOperator.NONE.suffix} on type ${type.name} is deprecated, use ${fieldName}${RelationOperator.NOT.suffix} instead")
```

Comment:
why don't we handle the other cases? i.e. with an error?

Reply:
b/c it is not an error